File: app/src/main/java/com/parabola/newtone/mvp/view/ArtistView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Album;
import com.parabola.domain.settings.ViewSettingsInteractor.AlbumViewType;
import com.parabola.newtone.mvp.CustomAddToEndSingleStrategy;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.StateStrategyType;

@StateStrategyType(CustomAddToEndSingleStrategy.class)
public interface ArtistView extends MvpView {
    void setArtistName(String artistName);
    void setTracksCount(int tracksCount);
    void setAlbumsCount(int albumsCount);

    void refreshAlbums(List<Album> albums);

    void setViewType(AlbumViewType viewType);
}

File: app/src/main/java/com/parabola/newtone/mvp/view/ArtistView.kt
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Album;
import com.parabola.domain.settings.ViewSettingsInteractor.AlbumItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface ArtistView extends MvpView {
    void setArtistName(String artistName);
    void setTracksCount(int tracksCount);
    void setAlbumsCount(int albumsCount);

    void refreshAlbums(List<Album> albums);

    void setAlbumViewSettings(AlbumItemView albumViewSettings);
    void setItemDividerShowing(boolean showed);

}
